src/modules/storage/LedgerStorage.ts

```diff
@@ -14,20 +14,65 @@
 
 *******************************************************************************/
 
+import * as sqlite from 'sqlite3';
+import { Hash }  from '../common/Hash'
 import { Storages } from './Storages';
 
 /**
  * The class that insert and read the ledger into the database.
  */
 export class LedgerStorage extends Storages
 {
+    private hash: Hash;
+
+    constructor (filename: string, callback: (err: Error | null) => void)
+    {
+        super(filename, callback);
+        this.hash = new Hash();
+    }
+
     /**
      * Creates tables related to the ledger.
      * @param callback If provided, this function will be called when
      * the database was finished successfully or when an error occurred.
      * The first argument is an error object.
      */
     public createTables (callback: (err: Error | null) => void)
+    {
+        this.createBlocksTable((err1: Error | null) =>
+        {
+            if (err1)
+            {
+                callback(err1);
+                return;
+            }
+
+            this.createTransactionsTable((err2: Error | null) =>
+            {
+                if (err2)
+                {
+                    callback(err2);
+                    return;
+                }
+
+                this.createInputsTable((err3: Error | null) =>
+                {
+                    if (err3)
+                    {
+                        callback(err3);
+                        return;
+                    }
+
+                    this.createOutputsTable((err4: Error | null) =>
+                    {
+                        callback(err4);
+                    });
+                });
+            });
+        });
```

Comment:
Why the split ? This makes the code worse for no reason.

Reply:
I used another method, and it didn't work.
I'll have to do it this way.

Reply:
I handled this in the way @TrustHenry  applied.

Reply:
I changed. Thanks.
``` TypeScript

    public createTables (callback: (err: Error | null) => void)
    {
        var sql: string =
        `CREATE TABLE IF NOT EXISTS blocks
        (
            height INTEGER NOT NULL PRIMARY KEY,
            prev_block TEXT NOT NULL,
            validators TEXT NOT NULL,
            merkle_root TEXT NOT NULL,
            signature TEXT,
            tx_count INTEGER NOT NULL,
            enrollment_count INTEGER NOT NULL
        );

        CREATE TABLE IF NOT EXISTS enrollments
        (
            block_height INTEGER NOT NULL,
            enrollment_index INTEGER NOT NULL,
            utxo_key TEXT NOT NULL,
            random_seed TEXT NOT NULL,
            cycle_length INTEGER NOT NULL,
            enroll_sig TEXT NOT NULL,
            PRIMARY KEY("block_height","enrollment_index")
        );

        CREATE TABLE IF NOT EXISTS "transactions"
        (
            "block_height"      INTEGER NOT NULL,
            "tx_index"          INTEGER NOT NULL,
            "tx_hash"           TEXT NOT NULL,
            "type"              INTEGER NOT NULL,
            "inputs_count"      INTEGER NOT NULL,
            "outputs_count"     INTEGER NOT NULL,
            PRIMARY KEY("block_height", "tx_index")
        );

        CREATE TABLE IF NOT EXISTS "tx_inputs"
        (
            "block_height"      INTEGER NOT NULL,
            "tx_index"          INTEGER NOT NULL,
            "in_index"          INTEGER NOT NULL,
            "previous"          TEXT NOT NULL,
            "out_index"         INTEGER NOT NULL,
            PRIMARY KEY("block_height", "tx_index", "in_index")
        );

        CREATE TABLE IF NOT EXISTS "tx_outputs"
        (
            "block_height"  INTEGER NOT NULL,
            "tx_index"      INTEGER NOT NULL,
            "output_index"  INTEGER NOT NULL,
            "tx_hash"       TEXT NOT NULL,
            "utxo_key"      TEXT NOT NULL,
            "amount"        NUMERIC NOT NULL,
            "address"       TEXT NOT NULL,
            "used"          INTEGER NOT NULL DEFAULT 0,
            PRIMARY KEY("block_height", "tx_index", "output_index")
        );

        CREATE TABLE IF NOT EXISTS "validators"
        (
            "enrolled_at"	INTEGER NOT NULL,
            "utxo_key"	TEXT NOT NULL,
            "address"	TEXT(56) NOT NULL,
            "amount"	NUMERIC NOT NULL,
            "preimage_distance"	INTEGER NOT NULL,
            "preimage_hash"	TEXT NOT NULL,
            PRIMARY KEY("enrolled_at","utxo_key")
        );
        `;

        this.db.exec(sql, (err: Error | null) =>
        {
            callback(err);
        });
    }
```